config/ubuntu/waagent.conf

```diff
@@ -21,7 +21,7 @@ Provisioning.RegenerateSshHostKeyPair=n
 Provisioning.SshHostKeyPairType=rsa
 
 # Monitor host name changes and publish changes via DHCP requests.
-Provisioning.MonitorHostName=n
+Provisioning.MonitorHostName=y
```

Comment:
Why change the default config for Ubuntu? It should be sufficient for the image maintainers to change the behavior of the shipped config in the package as opposed to us changing default behavior upstream.

Reply:
Then why are there config files for distros in the first place? (just wondering)

I assumed that the config is disabled just because it is not working for Ubuntu 18+. 
So I proposed to change that to y since itt is working fine with this patch.

I would exclude this change if required. Please suggest.

Reply:
Let's exclude the change. I don't want to change this default from upstream, as it isn't necessary given the scope of the PR.